File: ClientApp/src/app/core/enums/app.enums.ts
export enum UserRole {
    Visitor = 'Visitor',
    Anonymous = 'Anonymous',
    User = 'User',
    Admin = 'Admin',
    Expert = 'Expert',
    Reviewer = 'Reviewer',
    Author = 'Author',
    Instructor = 'Instructor',
    Mechanic = 'Mechanic',
    GarageOwner = 'GarageOwner',
    Vendor = 'Vendor',
    Affiliate = 'Affiliate',
}

export enum PostType {
    Text = 'text',
    Image = 'image',
    Video = 'video',
    Poll = 'poll',
    Article = 'article',
}

export enum NotificationType {
    Like = 'like',
    Comment = 'comment',
    Follow = 'follow',
    Mention = 'mention',
    System = 'system',
    Achievement = 'achievement',
}

export enum ToastType {
    Success = 'success',
    Error = 'error',
    Warning = 'warning',
    Info = 'info',
}

export enum Theme {
    Light = 'light',
    Dark = 'dark',
    Auto = 'auto',
}

export enum Language {
    English = 'en',
    Arabic = 'ar',
    Spanish = 'es',
    French = 'fr',
}

export enum SortOrder {
    Ascending = 'asc',
    Descending = 'desc',
}

export enum PostSortBy {
    Recent = 'recent',
    Trending = 'trending',
    Popular = 'popular',
    Oldest = 'oldest',
}

export enum ContentStatus {
    Draft = 'draft',
    Published = 'published',
    Archived = 'archived',
    Deleted = 'deleted',
}

export enum ModalSize {
    Small = 'sm',
    Medium = 'md',
    Large = 'lg',
    ExtraLarge = 'xl',
}

export enum LoadingState {
    Idle = 'idle',
    Loading = 'loading',
    Success = 'success',
    Error = 'error',
}

export enum HttpMethod {
    GET = 'GET',
    POST = 'POST',
    PUT = 'PUT',
    PATCH = 'PATCH',
    DELETE = 'DELETE',
}

export enum FileType {
    Image = 'image',
    Video = 'video',
    Document = 'document',
    Audio = 'audio',
}

export enum ActivityType {
    PostCreated = 'post_created',
    PostLiked = 'post_liked',
    CommentAdded = 'comment_added',
    UserFollowed = 'user_followed',
    ProfileUpdated = 'profile_updated',
    AchievementUnlocked = 'achievement_unlocked',
}

export enum AnalyticsPeriod {
    Today = 'today',
    Week = 'week',
    Month = 'month',
    Year = 'year',
    All = 'all',
}

export enum TrendDirection {
    Up = 'up',
    Down = 'down',
    Stable = 'stable',
}

export enum DeviceType {
    Desktop = 'desktop',
    Mobile = 'mobile',
    Tablet = 'tablet',
}

export enum BrowserType {
    Chrome = 'chrome',
    Firefox = 'firefox',
    Safari = 'safari',
    Edge = 'edge',
    Other = 'other',
}

File: src/Application/Common/Enums/AppEnums.cs
namespace Application.Common.Enums;

public enum UserRole
{
    Visitor = 0,
    Anonymous = 1,
    User = 2,
    Admin = 3,
    Expert = 4,
    Reviewer = 5,
    Author = 6,
    Instructor = 7,
    Mechanic = 8,
    GarageOwner = 9,
    Vendor = 10,
    Affiliate = 11
}

public enum PostType
{
    Text = 0,
    Image = 1,
    Video = 2,
    Poll = 3,
    Article = 4
}

public enum NotificationType
{
    Like = 0,
    Comment = 1,
    Follow = 2,
    Mention = 3,
    System = 4,
    Achievement = 5
}

public enum ContentStatus
{
    Draft = 0,
    Published = 1,
    Archived = 2,
    Deleted = 3
}

public enum SortOrder
{
    Ascending = 0,
    Descending = 1
}

public enum PostSortBy
{
    Recent = 0,
    Trending = 1,
    Popular = 2,
    Oldest = 3
}

public enum ActivityType
{
    PostCreated = 0,
    PostLiked = 1,
    CommentAdded = 2,
    UserFollowed = 3,
    ProfileUpdated = 4,
    AchievementUnlocked = 5
}

public enum AnalyticsPeriod
{
    Today = 0,
    Week = 1,
    Month = 2,
    Year = 3,
    All = 4
}

public enum TrendDirection
{
    Up = 0,
    Down = 1,
    Stable = 2
}

public enum FileType
{
    Image = 0,
    Video = 1,
    Document = 2,
    Audio = 3
}

public enum DeviceType
{
    Desktop = 0,
    Mobile = 1,
    Tablet = 2
}

public enum BrowserType
{
    Chrome = 0,
    Firefox = 1,
    Safari = 2,
    Edge = 3,
    Other = 4
}

public enum EmailTemplate
{
    Welcome = 0,
    PasswordReset = 1,
    EmailVerification = 2,
    NewFollower = 3,
    NewComment = 4,
    WeeklyDigest = 5
}

public enum CacheStrategy
{
    None = 0,
    Short = 1,
    Medium = 2,
    Long = 3,
    Permanent = 4
}
